Glasgow - Class 6 - Mandy Cheung - React - Hotel Group Project

m4ndycheung opened the pull request:
This is buddy group g's pull request for the React Hotel Project.

Buddies: Cat, Haroun, Heresh, Mandy
Buddy mentor: Douglas

HereshT committed:
Extracted the search <button> from the src/Search.js file

New component file SearchButton.js has been completed.

Updated  Search.js component to use this SearchButton component.

HereshT committed:
Heading.js component has been created.

A new component file Heading.js has been created.

I have updated App.js component to use this Heading component.

A HotelIcon SVG component has been created.

HereshT committed:
Merge pull request #2 from m4ndycheung/baranch-HereshT

Extracted the search <button> from the src/Search.js filee

m4ndycheung committed:
Created TouristInfoCards Component

Added some data for the cards

m4ndycheung committed:
Linked TouristInfoCards component to App.js

HereshT committed:
Lesson 1 task 5. Create a table to show hotel bookings is Done.

I have Created a SearchResults  component that shows hotel bookings in a <table> element. Each booking has an id, title, first name, surname, email, room id, check in date and check out date.

m4ndycheung committed:
Created 3 x tourist info cards

Not added the css yet.

m4ndycheung committed:
Added link url to button

HereshT committed:
A column that shows number of nights for each booking

Another column has been added to  <SearchResults /> table which shows the number of nights a guest is staying.

m4ndycheung committed:
added flex to cards

m4ndycheung committed:
Merge pull request #3 from m4ndycheung/Part-1-3-MandyC

Mandy C - Lesson 1.3. Create and use a new component to show info cards

hanl00 reviewed (commented):
Looks good, no glaring errors here

hanl00 commented on src/TouristInfoCards.js:
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+
+function TouristInfoCards() {
```
I would personally stick to one way to declare react functional components, so this is more of a consistency/nitpick comment

https://github.com/CodeYourFuture/cyf-hotel-react/pull/573#discussion_r1215415298

m4ndycheung committed:
Merge branch 'master' into baranch-HereshT

m4ndycheung committed:
Merge pull request #4 from m4ndycheung/baranch-HereshT

A column that shows the number of nights for each booking

m4ndycheung committed:
Added keys to tourist info cards

m4ndycheung committed:
Merge pull request #5 from m4ndycheung/Part-1-3-MandyC

Lesson 1 Part 3

m4ndycheung committed:
Corrected dependencies

Douglas helped me with this, there was a dependency called moment missing from my local copy. We have added this now.

m4ndycheung committed:
Created Footer component file

Added a hello world as a test

m4ndycheung committed:
Created list of contact details

m4ndycheung committed:
Had a go at destructuring props

m4ndycheung committed:
Added footer css file and added styling

m4ndycheung committed:
Rendered Restaurant component

m4ndycheung committed:
Completed  Lesson 2.9 Preparing to add more pizzas

m4ndycheung committed:
We can add more pizzas now

m4ndycheung committed:
Created restaurant button component

m4ndycheung committed:
Made amendments following code review

Changed to camelCase for orderOne function.

Updated orderOne function with more robust code.

m4ndycheung committed:
Updated naming

HereshT committed:
Lesson 2 > Tasks 14 , 15 are completed.

 Lesson 2 > Tasks 14.  Passing bookings from a state variable is completed.

 Lesson 2 > Tasks 15. Highlight booking row when clicked.

+

Plus a few small CSS changes.

HereshT committed:
Update Bookings.js

HereshT committed:
Lesson 2 part 16. Load bookings remotely is Done.

Lesson 2 part 16. Load bookings remotely, now we will get and load the bookings from a remote API.

I have added Restaurant.css

HereshT committed:
Update Bookings.js

HereshT committed:
Merge pull request #12 from m4ndycheung/baranch-HereshT

 Lesson 2 > Tasks 14 , 15 are completed.

HereshT committed:
Update Bookings.js

HereshT committed:
Updated Bookings.js

HereshT committed:
Merge branch 'master' into baranch-HereshT

HereshT committed:
Merge pull request #13 from m4ndycheung/baranch-HereshT

Updated Bookings.js

HereshT committed:
Lesson 2 part 16. Load bookings remotely  has been updated

HereshT committed:
Merge pull request #14 from m4ndycheung/baranch-HereshT

Lesson 2 part 16. Load bookings remotely  has been updated

m4ndycheung committed:
Created Order component

m4ndycheung committed:
Passed props to Order component

HereshT committed:
Triggering search when submitting the form

Showing text that is typed in the search input field when submitting the form in the console.

HereshT committed:
Triggering search when submitting the form

Showing text that is typed in the search input field when submitting the form in the console.

HereshT committed:
Implementing the search functionality

HereshT committed:
Implementing the search functionality

HereshT committed:
Task 22. Show a loading message

A message inviting the user to wait has been displayed on the screen until bookings data can be rendered on the screen, and then when bookings are rendered, the loading message is gone.

HereshT committed:
Update Bookings.js

HereshT committed:
Merge branch 'master' into baranch-HereshT

HereshT committed:
Task 23. Show an error message

When loading bookings data from the /error endpoint, an error message is displayed on the screen.

HereshT committed:
Task 23. Show an error message

When loading bookings data from the /error endpoint, an error message is being displayed on the screen.

HarounAlarabi committed:
I added 20/21 and updated TouristInfocard

HarounAlarabi committed:
Merge pull request #19 from HarounAlarabi/master

I added 20/21 and updated TouristInfocard

HarounAlarabi committed:
updated the style

HarounAlarabi committed:
Merge pull request #20 from HarounAlarabi/master

updated the style

HereshT committed:
Merge pull request #18 from m4ndycheung/baranch-HereshT

Task 22. Show a loading message

hanl00 commented on src/SearchResults.js:
```diff
@@ -0,0 +1,86 @@
+  return (
+    <tr onClick={toggleHighlight} className={highlightClass}>
+      <th scope="row">{booking.id}</th>
+      <td>{booking.title}</td>
+      <td>{booking.firstName}</td>
+      <td>{booking.surname}</td>
+      <td>{booking.email}</td>
+      <td>{booking.roomId}</td>
+      <td>{booking.checkInDate}</td>
+      <td>{booking.checkOutDate}</td>
```
you can destructure the booking object so you dont have to use `booking.` on every line

hanl00 commented on src/CustomerProfile.js:
```diff
@@ -0,0 +1,41 @@
+  }
+
+
+return(
+
+
+  <fieldset class="scheduler-border">
+    <button className="closeBtn"  onClick={onCloseProfile}>X</button>
+  <legend class="scheduler-border">Customer {id} Profile</legend>
+  {customerProfile && (
+  <ul class="control-group">
+  <li>Customer ID: {customerProfile.id}</li>
+          <li>Email: {customerProfile.email}</li>
+          <li>VIP: {customerProfile.vip ? "Yes" : "No"}</li>
+          <li>Phone Number: {customerProfile.phoneNumber}</li>
+  </ul>
+  )}
+
+</fieldset>
+);
+
+}
```
code looks good but isn't formatted right, i would suggest installing prettier code formatter on vs code extensions
